### DISCUSSION TITLE
[Ollama + OpenWebUI + Cloudflare Tunnel]: WebUI could not connect to Ollama

### DISCUSSION THREAD

**Discussion by matsyui** [open-webui/open-webui#5271] [General]
Just to make things clear there's a way using Cloudflare Tunnel to work and make api ollama connected with Open-WebUI by using this method [How can I use Ollama with Cloudflare Tunnel?](https://github.com/ollama/ollama/blob/main/docs/faq.md#how-can-i-use-ollama-with-cloudflare-tunnel):

`cloudflared tunnel --url http://localhost:11434 --http-host-header="localhost:11434"`

By using this method, you're sacrificing using your own domain name, which is not that good because it this will generate a quick Tunnel with this sub and domain:

`https://XXXXXXXXXXXXXXXXXXXXXXX.trycloudflare.com/`

Using this method will allow us to connect ollama api with CF Tunnels
![image](https://github.com/user-attachments/assets/f198b734-e6d0-498c-a465-3f2ebe86f578)

But with the method making it more personalized and have control with the tunnel, Creating your own URL with the same setup HTTP, localhost and port and the HTTP host header
![image](https://github.com/user-attachments/assets/8f5a9241-9d8d-4e09-941c-a323952e4f68)

It will not work, same config/settings but it just don't work.
![image](https://github.com/user-attachments/assets/3ba70c70-6674-4ce8-92ab-1a2acb5c9033)

But, using the URL cloudflared via browser and other API tools, it is working just fine. Here's an example output on Browser and Thunder Client:
![image](https://github.com/user-attachments/assets/0af7adff-6d9d-4f3f-8b7e-4bd0f50fdfee)

The reason i want to push this tunnel because I noticed multiple times that the response from tunneled is much faster than direct/local. But also I want it to have more control over the tunnel configuration and a personalized url for my Ollama's api
![image](https://github.com/user-attachments/assets/1f87aecb-24f4-4698-ad8b-c20090f6f3f4)

**Comment by yuchenxie4645** [comment]
I do not think you can tunnel ollama api over cloudflare proxy like that. In your DNS, make sure to NOT have cloudflare PROXY the IP. If cloudflare proxies the IP it would go through the cloudflare server then to host. Disabling cloudflare proxy will directly forward the connection to your ip.

**Reply by yuchenxie4645** [reply]
Additionally, if you just put the domain (https://ollama.x.x) without the :11434, it also woudlnt work, as you set ollama port to 11434. You can obviously change that setting in the original rule of cloudflare.

**Reply by matsyui** [reply]
I have updated the entire post for you and added docs source of ollama using a cloudflare tunnel.

**Reply by yuchenxie4645** [reply]
It looks like you hosted Ollama on a separate machine than openwebui and you want to bridge these two using cloudflare tunnel. I also see you mention that it works on browser and other API endpoints, maybe in your settings, try it unproxied. Logically, it should've worked, and I feel like you did everything correctly, and it was Openwebui's fault. So I would try if unproxied through CF tunnel would work.
